doc/aaa/SONiC RBAC.md

```diff
@@ -0,0 +1,486 @@
+- `operator` users will be added to `docker` group only.
+- All non-admin users will be added to the `docker` group only.
+
+The default shell for all users is set to a script. This script will determine
+if the user is a member of the admin group, and if so, drop them into a Bash
+shell, otherwise, it will spawn KLISH to drop them into the CLI. Non-admin users
+will only have access to the CLI and not be able to access Bash. Users cannot
+run `system` commands from KLISH.
+
+#### 1.1.1.5 Customizable roles
+
+Admin users shall be able to create customizable roles for further granularity beyond read-only and read-write. For instance, a `secadmin` role shall allow the users to configure security parameters on the switch. The custom role shall have a default deny policy. Custom roles shall map a set of features, and their access level to the feature. A feature is a set of YANG paths. The default policy is no access to a feature, if it is not specified in the role.
+
+```
+[
+  {
+    "name": "secadmin",
+    "features": [
+        {
+            "feature": "aaa",
+            "access": "rw"
+        },
+        {
+            "feature": "net",
+            "access": "ro"
+        }
+    ]
+  }
+]
+```
+
+Features map to a set of YANG paths. The features are saved in the
+ResourceTable, but a default feature set is defined in the system to allow users
+to start from a known state. Sample feature description is shown below.
+
+```
+[
+    {
+        "name": "aaa",
+        "paths": [
+            "openconfig-aaa:aaa/system/users/user={username}",
+            "openconfig-aaa:aaa/system/users"
+        ]
+    },
+    {
+        "name": "net",
+        "paths": [
+            "openconfig-net:net/xyz"
+        ]
+    }
+]
+```
+
+
+#### 1.1.1.6 Local User Management and UserDB Sync
+An interface must be developed for local user management, so that administrators can add users and assign passwords and roles to them. Administrators with the appropriate role must be able to add/delete users, modify user passwords, and modify user roles. They must be able to do so through all of the NBIs, meaning that a YANG model and CLI tree must be developed.
+
+Users must be added to the Linux database (`/etc/passwd`, `/etc/group`, and optionally `/etc/shadow`). That's where a user is mapped to a Linux [User Identifier](https://en.wikipedia.org/wiki/User_identifier) (UID) and primary [Group Identifier](https://en.wikipedia.org/wiki/Group_identifier) (GID). When users are created they also need to be assigned roles. Roles are simply defined as Linux groups (`/etc/group`) and assigned to users as [Supplementary GIDs](https://en.wikipedia.org/wiki/Group_identifier#Supplementary_groups).
+
+When a user is created it also needs to be assigned certificates that will allow them to communicate with the REST server. Finally, all users need to be added to the REDIS database where additional information about each user can be stored.
```

Comment:
Which certificates need to be assigned? I believe the KLISH was supposed to communicate using User Roles per the new design. The certgen utility appears to have been removed.

Reply:
This can be taken out, since we no longer need to use certificate based authentication for CLI.